### ISSUE TITLE
publishWithContext will blocked forever when connection bloced

### ISSUE THREAD

**Issue by jqiris** [rabbitmq/amqp091-go#111]
publishWithContext will blocked forever when connection bloced

**Comment by Zerpet** [comment]
That is correct. Connections can get blocked for some reason, mainly alarms and flow control, and RabbitMQ will stop reading from the TCP socket. You can subscribe to `connection.blocked` and `connection.unblocked` events using [Connection.NotifyBlocked](https://pkg.go.dev/github.com/rabbitmq/amqp091-go#Connection.NotifyBlocked) to react to those events.

**Comment by Zerpet** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(
